unified-codes/apps/web/src/components/containers/app/AppPage.tsx

```diff
@@ -20,29 +16,51 @@ const useStyles = makeStyles((theme: ITheme) => createStyles({
    },
 }));
 
-export const AppPage = () => {
+export interface AppPageProps {
+    onMount: () => void;
+    onUnmount: () => void;
+}
+
+export type AppPage = React.FunctionComponent<AppPageProps>;
+
+export const AppPage = ({ onMount = () => null, onUnmount = () => null }) => {
     const classes = useStyles();
 
+    const { 
+        DETAIL: DetailRoute,
+        EXPLORER: ExplorerRoute,
+        LOGIN: LoginRoute,
+        ERROR: ErrorRoute,
+    } = ROUTES;
+
+    const {
+        DEFAULT: defaultPath,
+        DETAIL: detailPath,
+        EXPLORER: explorerPath,
+        LOGIN: loginPath,
+        ERROR: errorPath
+    } = PATHS;
+
     return (
         <Box className={classes.root}>
             <Switch>
-                <Route path="/detail/:code">
-                    <DetailPage />
+                <Route exact path={defaultPath}>
+                    <ExplorerRoute />
                 </Route>
-                <Route exact path="/">
-                    <ExplorerPage />
+                <Route exact path={detailPath}>
+                    <DetailRoute />
                 </Route>
-                <Route exact path="/explorer">
-                    <ExplorerPage />
+                <Route exact path={explorerPath}>
+                    <ExplorerRoute />
                 </Route>
-                <Route exact path="/login">
-                    <Login />
+                <Route exact path={errorPath}>
+                    <ErrorRoute />
                 </Route>
-                <Route exact path="/404">
-                    <ErrorPage code="404" />
+                <Route exact path={loginPath}>
+                    <LoginRoute />
                 </Route>
                 <Route>
-                    <Redirect to="/404" />
+                    <Redirect to={errorPath} />
```

Comment:
This one should be redirecting to `/error/404` - as it stands the user gets routed to the URL `http://localhost:4200/error/:code`

Reply:
3b2705216e6e80f418c76bfa4ee81f2d4834e3c7.